lib/src/protocol/http/parser/mod.rs

```diff
@@ -278,7 +279,55 @@ fn message_header(i: &[u8]) -> IResult<&[u8], Header> {
         crlf,
     ))(i)?;
 
-    Ok((i, Header { name, value }))
+    Ok((
+        i,
+        Header {
+            name: name.to_owned(),
+            value: value.to_owned(),
+        },
+    ))
+}
+
+#[cfg(not(feature = "tolerant-http1-parser"))]
+fn message_header(i: &[u8]) -> IResult<&[u8], Header> {
+    // ToDo handle folding?
+    let (i, (name, _, _, value, _)) = tuple((
+        token,
+        tag(":"),
+        take_while(is_space),
+        take_while(is_header_value_char),
+        crlf,
+    ))(i)?;
+
+    // header value are formatted using character in the ascii table, so we are
+    // sure that the utf8 conversion will not fail.
+    //
+    // According to the rfc 2616, section 2.2 and 4.2, the linear white space
+    // (lws) which are the horizontal tabulation ('\t') and the space (' ') MAY
+    // be trimmed.
+    //
+    // See rfc 2616, section 4.2:
+    //
+    // > The field-content does not include any leading or trailing LWS:
+    // > linear white space occurring before the first non-whitespace
+    // > character of the field-value or after the last non-whitespace
+    // > character of the field-value. Such leading or trailing LWS MAY be
+    // > removed without changing the semantics of the field value. Any LWS
+    // > that occurs between field-content MAY be replaced with a single SP
+    // > before interpreting the field value or forwarding the message
+    // > downstream.
+    let value = String::from_utf8_lossy(value)
```

Comment:
there is probably a way to trim without allocating, by iterating from the last character of the slice an only checking for space characters (you will even avoid the utf8 parsing).

It is also the wrong place to trim: `Transfer-Encoding` can get a comma separated list of values, so if we get the header `Transfer-Encoding: chunked\t, gzip\r\n`, the bug will still be present because this only trims at the end of the header value